craft_parts/executor/part_handler.py

```diff
@@ -246,6 +277,14 @@ def _run_overlay(
                     stderr=stderr,
                 )
 
+                # apply overlay-organize
+                organize_files(
+                    part_name=self._part.name,
+                    file_map=self._part.spec.overlay_organize_files,
+                    install_dir_map=self._part.part_layer_dirs,
```

Comment:
Since we have a single overlay_manager a mapping with the only valid source is the overlay_mount_dir and the destinations are the layer dirs of the part in the partitions.

Reply:
Implemented this in a slightly easier to read way.

Reply:
Also fix the corresponding test to organize from the assembled overlay stack.